specification/data-resource-semantic-conventions.md

```diff
@@ -70,6 +70,23 @@ Labels defining a compute unit (e.g. Container, Process, Lambda Function).
 | container.image.name | Name of the image the container was built on. | `gcr.io/opentelemetry/operator` |
 | container.image.tag | Container image tag. | `0.1` |
 
+
+### Function as a Service
+
+**type:** `faas`
+
+**Description:** An instance of a function that runs on a platform where underlying hardware concerns, such as provisioning and scaling, have been abstracted away by the platform.
+
+| Label  | Description  | Example  | Required? |
+|---|---|---|--|
+| faas.name | The name of the function being executed. | `my-function` | Yes |
+| faas.identifier | The unique name of the function being executed. <br /> For example, in AWS Lambda this field corresponds to the [ARN] value, in GCP to the URI of the resource, and in Azure to the [FunctionDirectory] field. | `arn:aws:lambda:us-west-2:123456789012:function:my-function` | Yes |
+| faas.service | Platform name where the function is executed, e.g. `aws-lambda`, `cloud-functions`, `azure-functions`, ... | `aws-lambda` | Yes |
```

Comment:
Propose some way to make sure these are as consistent as possible. Possibly a list of known values so far? It should be flexible enough to allow a new service we don't currently know about to be used here, but it should not be so flexible that different SIGs use different names for the same service. For instance, one SIG may put `cloud-functions` and another might put `google-cloud-functions`.

Reply:
Looking at the resource list [(link)](https://github.com/dynatrace-oss-contrib/opentelemetry-specification/blob/master/specification/data-resource-semantic-conventions.md#cloud), there is the *cloud* convention which partially cover the purpose of this attribute. However, I am thinking that this level of granularity is not needed given the problems you mentioned.

Reply:
Are you planning to change this then? E.g. move it to "cloud.service" or something like that?

Reply:
I am more keen on completely remove it. The original idea was to allow services that are build on top of FaaS (e.g. serverless, claudia.js, or even the multiple different implementations offered by azure) to tag the resource with their own identifier. I am not sure anymore that this fine grained level is necessary.

Reply:
Agreed, let's remove that.